### DISCUSSION TITLE
Instead of test error make test fail if during test step has encounters an exception

### DISCUSSION THREAD

**Discussion by f00dog-x300** [pytest-dev/pytest#10378] [Q&A]
Hello, 

So I have code that looks something like this: 

```
def do_action() -> None:
    # tries to do something
  

def test_that_action:
    get_result_of_action = do_action()
    # do some morphing of the data
    # make final assertion that end result is what we expect
```

If an exception occurs during the `do_action` function, pytest marks the test as an error. I think that's expected. However, is there a way to set it up in pytest where if an exception is encountered during the test that if a function call inside of the test encounters an exception will cause the test to fail?

**Comment by nicoddemus** [comment]
Hi,

Not currently. 

I think you can override the [pytest_runtest_makereport](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_runtest_makereport) to change it from an error to a failure though.
